BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -523,7 +557,143 @@ open class SBAUserDefaultsProfileItem: SBAProfileItemBase {
+        guard let studyParticipant = SBAStudyParticipantProfileItem.studyParticipant
+            else {
+                assertionFailure("Attempting to read \(key) (\(profileKey)) on nil SBBStudyParticipant")
+                return nil
+        }
+        guard let enumKey = SBAProfileParticipantSourceKey(rawValue: key)
+            else {
+                assertionFailure("Error reading \(key) (\(profileKey)): \(key) is not a valid SBBStudyParticipant key")
+                return nil
+        }
+        
+        var storedVal = studyParticipant.value(forKey: key)
+        switch enumKey {
+        case .sharingScope:
+            guard let scopeString = storedVal as? String else { break }
+            storedVal = SBBParticipantDataSharingScope(key: scopeString)
+        default:
+            break
+        }
+        
+        return storedVal
+    }
+    
+    override open func setStoredValue(_ newValue: Any?) {
+        guard let studyParticipant = SBAStudyParticipantProfileItem.studyParticipant
+            else {
+                assertionFailure("Attempting to set \(sourceKey) (\(profileKey)) on nil SBBStudyParticipant")
+                return
+        }
+        guard let key = SBAProfileParticipantSourceKey(rawValue: sourceKey)
+            else {
+                assertionFailure("Error setting \(sourceKey) (\(profileKey)): \(sourceKey) is not a valid SBBStudyParticipant key")
+                return
+        }
+        
+        var setValue = newValue
+        switch key {
+        case .dataGroups:
+            guard let _ = newValue as? Set<String>
+                else {
+                    assertionFailure("Error setting \(sourceKey) (\(profileKey)): value \(String(describing: newValue)) cannot be converted to Set")
+                    return
+            }
+        case .sharingScope:
+            guard let scope = newValue as? SBBParticipantDataSharingScope
+                else {
+                    assertionFailure("Error setting \(sourceKey) (\(profileKey)): value \(String(describing: newValue)) cannot be converted to SBBParticipantDataSharingScope")
+                    return
+            }
+            setValue = SBBParticipantManager.dataSharingScopeStrings()[scope.rawValue]
+        case .notifyByEmail:
+            guard let _ = newValue as? Bool
+                else {
+                    assertionFailure("Error setting \(sourceKey) (\(profileKey)): value \(String(describing: newValue)) cannot be converted to Bool")
+                    return
+            }
+        default:
+            // the rest are String, and anything can be converted to String
+            break
+        }
+        
+        studyParticipant.setValue(setValue, forKeyPath: sourceKey)
+        
+        // save the change to Bridge
+        SBABridgeManager.updateParticipantRecord(studyParticipant) { (_, _) in }
+    }
+}
+
+
+open class SBAStudyParticipantCustomAttributesProfileItem: SBAProfileItemBase {
```

Comment:
Can you add an example for this to the json? phone and contact email would both be of this type, yeah?